csharp/src/Apache.Arrow/Arrays/Date32Array.cs

```diff
@@ -57,10 +57,10 @@ public Date32Array(ArrayData data)
 
         public DateTimeOffset? GetDate(int index)
         {
-            var values = ValueBuffer.Span.CastTo<int>().Slice(0, Length);
-            var value = values[index];
-
-            return DateTimeOffset.FromUnixTimeMilliseconds(value * MillisecondsPerDay);
+            var value = GetValue(index);
+            return value.HasValue ?
```

Comment:
I wonder if it wouldn't be simpler to just write:

```C#
if (!value.HasValue)
{
    return value;  // or `return default;`
}

return DateTimeOffset.FromUnixTimeMilliseconds(value.Value * MillisecondsPerDay);
```

Reply:
I agree. fix it.